x/shield/keeper/purchase.go

```diff
@@ -87,6 +87,12 @@ func (k Keeper) purchaseShield(ctx sdk.Context, poolID uint64, shield sdk.Coins,
 		return types.Purchase{}, types.ErrPoolInactive
 	}
 
+	// Check if the amount of purchased shield is to small.
+	poolParams := k.GetPoolParams(ctx)
+	if poolParams.MinShieldPurchase.IsAnyGT(shield) {
```

Comment:
Maybe only check for PurchaseShield? We want to be flexible for Create/Update pool

Reply:
Updated.